### PR TITLE
Fixed to handle manifest parse errors

### PR DESCRIPTION
**Pull request by vallentin**
[[sc-65915](https://app.shortcut.com/connectedcars/story/65915/rust-checks-are-not-skipped-if-cargo-toml-has-issues)]

### CODE REVIEW

File: src/checks/cargo/cargo.ts [hunk 1]
```diff
@@ -96,7 +115,14 @@ function getPrimaryOrFirstSpan(spans: DiagnosticSpan[]): DiagnosticSpan | null {
   return primarySpan ?? spans[0]
 }
 
-export function cargoHasBuildFinished(output: CargoMessage[]): boolean {
+export function cargoHasBuildFinishedSuccessfully(output: CargoMessage[]): boolean {
```
Comment (Ratoone) [inline src/checks/cargo/cargo.ts:118]: The name is kinda long, I suggest something like `cargoBuildsSuccessfully` or similar

Comment (vallentin) [reply src/checks/cargo/cargo.ts:118]: One issue with that name, is that it doesn't suggest a boolean is returned, where functions usually tell that by using `is` and `has`

Comment (Ratoone) [reply src/checks/cargo/cargo.ts:118]: Then `isCargoBuildSuccess` or `isCargoBuiltSuccessfully`

Comment (vallentin) [reply src/checks/cargo/cargo.ts:118]: Now it just sounds like broken English :)

Comment (Ratoone) [reply src/checks/cargo/cargo.ts:118]: Not if you write `isCargoBuildSuccessfully` - referring to `cargo build`

**Comment by vallentin** [comment]
Only rebased, no need to review
Will merge, when checks finish

**Pull request merged**